crates/vhost/src/vhost_user/slave_req.rs

```diff
@@ -126,7 +128,7 @@ impl SlaveFsCacheReq {
     }
 }
 
-impl VhostUserMasterReqHandler for SlaveFsCacheReq {
+impl VhostUserMasterReqHandler for Slave {
```

Comment:
Since we are here, should we factor out `fs_*` methods from the `VhostUserMasterReqHandler` trait?

Reply:
This is how `Slave` and `Master` look after this change:

`
pub struct Slave {
    // underlying Unix domain socket for communication
    node: Arc<Mutex<SlaveInternal>>,
}

pub struct Master {
    node: Arc<Mutex<MasterInternal>>,
}
`

And so I went with just `Slave` here.

Reply:
I am not sure why we should do that.

`VhostUserSlaveReqHandler` trait defines the helpers required by the backend to process messages from frontend.

Similarly, `VhostUserMasterReqHandler` trait defines the helpers required by the frontend to process messages from backend. And the `FS_*` messages should be handled from within `VhostUserMasterReqHandler`.

Is my understanding correct ?

Reply:
I'm confused, I interpreted that `fs_*` functions should be implemented only by backends that implement a `virtio-fs` device (e.g. virtiofsd)

Reply:
Yes that is correct, but they still must be part of VhostUserMasterReqHandler, that is where all requests from backend are collected. Many of the requests are optional and are only required to be implemented by the specific use cases, like virtiofsd in this case.

Reply:
Okay, got it!